### PR TITLE
update onNotificationsChange to include payload

### PR DESCRIPTION
**Pull request by riquelme-stripe**
Adding payload to onNotificationsChange

### CODE REVIEW

**Comment by riquelme-stripe** [comment]
https://github.com/stripe/react-connect-js/assets/170659229/e8966991-bab5-498b-975e-24b6a89ab9f1

File: src/Components.tsx [hunk 1]
```diff
@@ -38,6 +38,11 @@ export type CollectionOptions = {
   futureRequirements?: 'omit' | 'include';
 };
 
+export type NotificationCount = {
```
Comment (jorgea-stripe) [inline src/Components.tsx:41]: Instead of defining the type here again, can you import it from `connect-js`?

**Review by jorgea-stripe — CHANGES_REQUESTED** [review]
LGTM except for the duplicate type!

File: package.json [hunk 2]
```diff
@@ -49,7 +49,7 @@
     "@babel/preset-react": "7.18.6",
     "@rollup/plugin-json": "^6.0.0",
     "@rollup/plugin-replace": "^2.3.1",
-    "@stripe/connect-js": "3.3.7",
```
Comment (jorgea-stripe) [inline package.json:52]: The `peerDependencies` entry below should be bumped as well

**Review by jorgea-stripe — CHANGES_REQUESTED** [review]

File: package.json (hunk 2 above)
Comment (riquelme-stripe) [reply package.json:52]: Thanks for catching that

**Review by hliana-stripe — APPROVED** [review]

**Review by jorgea-stripe — APPROVED** [review]

**Pull request merged**